pennylane/qchem/tapering.py

```diff
@@ -538,3 +538,77 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
         tapered_hartree_fock.append(0)
 
     return np.array(tapered_hartree_fock).astype(int)
+
+
+def _is_commuting_obs(ham_a, ham_b, wire_map=None):
```

Comment:
I recommend checking with the operator arithmetic team to understand how they are thinking of checking commutativity between operators. We don't want to have duplicate functionality in the codebase

Reply:
@PennyLaneAI/core @albi3ro

Reply:
So we have a `qml.is_commuting` function, currently located in `ops/functions`.

The function isn't particularly general or easy to understand, but I do think we should merge both this logic and `qml.grouping.is_commuting` into that function.

Especially merging in ` qml.grouping.is_commuting`.  Rather confusing having two functions with the same names but different behaviours and locations in the code base.

Reply:
See Issue #3007 .  Core squad will try to combine `qml.grouping.is_commuting` and `qml.is_commuting` before the next release.